generator/src/test/java/com/scottlogic/deg/generator/cucumber/After.feature

```diff
@@ -0,0 +1,323 @@
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | 2019-01-01T00:00:00.000 |
+  | 2018-12-31T23:59:59.999 |
+  | 2018-12-31T23:59:59.998 |
+  | 2018-12-31T23:59:59.997 |
+  | 2018-12-31T23:59:59.996 |
+
+### Contradictions ###
+
+#Defect #672 - Request for contradictory After and Not After should only generate null
+@ignore
+Scenario: 'After' with a contradicting 'Not After Or At' only generates null
+Given foo is after 2019-01-01T00:00:00.000
+  And foo is anything but after or at 2019-01-01T00:00:00.000
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | null                    |
+
+
+### before ###
+
+  #Related to #667 - Expected time increment is one millisecond
+@ignore
+Scenario: 'After' with a non-contradicting 'Before' is successful
+Given foo is after 2019-01-01T00:00:00.000
+  And foo is before 2020-01-01T00:00:00.000
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | 2019-01-01T00:00:00.001 |
+  | 2019-01-01T00:00:00.002 |
+  | 2019-01-01T00:00:00.003 |
+  | 2019-01-01T00:00:00.004 |
+  | 2019-01-01T00:00:00.005 |
+
+# Defect 611 "before run against a non contradicting before results in an error" related to this scenario
+@ignore
+Scenario: 'After' with a non-contradicting 'Not Before' is successful
+Given foo is after 2019-01-01T00:00:00.000
+  And foo is anything but before 2019-01-01T00:00:00.000
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | 2019-01-01T00:00:00.001 |
+  | 2019-01-01T00:00:00.002 |
+  | 2019-01-01T00:00:00.003 |
+  | 2019-01-01T00:00:00.004 |
+  | 2019-01-01T00:00:00.005 |
+
+  # Defect 611 "before run against a non contradicting before results in an error" related to this scenario
+@ignore
+Scenario: 'Not After' with a non-contradicting 'Before' is successful
+Given foo is anything but after 2019-01-01T00:00:00.000
+  And foo is before 2019-01-01T00:00:00.000
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | 2018-12-31T23:59:59.999 |
+  | 2018-12-31T23:59:59.998 |
+  | 2018-12-31T23:59:59.997 |
+  | 2018-12-31T23:59:59.996 |
+  | 2018-12-31T23:59:59.995 |
+
+ #Related to #667 - Expected time increment is one millisecond
+@ignore
+Scenario: 'Not After' with a non-contradicting 'Not Before' is successful
+Given foo is anything but after 2019-01-01T00:00:00.000
+  And foo is anything but before 2018-01-01T00:00:00.000
+  And the generator can generate at most 5 rows
+Then the following data should be generated:
+  | foo                     |
+  | 2018-01-01T00:00:00.000 |
+  | 2018-01-01T00:00:00.001 |
+  | 2018-01-01T00:00:00.002 |
+  | 2018-01-01T00:00:00.003 |
+  | 2018-01-01T00:00:00.004 |
+
+### Contradictions ###
```

Comment:
is this comment repeated?

Reply:
Contradictions tests are subsets of combination sets

Reply:
Not sure this comment is required, as its normally mentioned in the scenario description (i.e. _with contradicting 'Not Before'..._)

Reply:
Removed